images/velero-plugin-for-csi/tests/docker-test.sh

```diff
@@ -0,0 +1,240 @@
+    echo "Velero pod is not running"
+  fi
+
+  # find the default storage class
+  defaultStorageClass=$(kubectl get storageclass -o jsonpath='{.items[?(@.metadata.annotations.storageclass\.kubernetes\.io/is-default-class=="true")].metadata.name}')
+  
+  # create the VolumeSnapshotClass
+  cat <<EOF | kubectl apply -f -
+    apiVersion: snapshot.storage.k8s.io/v1
+    kind: VolumeSnapshotClass
+    metadata:
+      name: local-path-snapshot-class
+    driver: $defaultStorageClass
+    deletionPolicy: Delete
+    parameters:
+      # Replace these with your MinIO S3 storage details and any additional CSI driver parameters
+      s3Url: "http://minio.velero.svc:9000" # Your MinIO service URL
+      s3BucketName: "velero" # Your MinIO bucket name for snapshots
+      s3Region: "us-east-1"
+      s3AccessKey: "minio" # Use Kubernetes secrets for production
+      s3SecretKey: "minio123" # Use Kubernetes secrets for production
+EOF
+
+# https://raw.githubusercontent.com/vmware-tanzu/velero/main/examples/nginx-app/with-pv.yaml
+  cat <<EOk > base.yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: nginx-example
+  labels:
+    app: nginx
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nginx-logs
+  namespace: nginx-example
+  labels:
+    app: nginx
+spec:
+  storageClassName: $defaultStorageClass
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 50Mi
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: nginx-example
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+      annotations:
+        pre.hook.backup.velero.io/container: fsfreeze
+        pre.hook.backup.velero.io/command: '["/sbin/fsfreeze", "--freeze", "/var/log/nginx"]'
+        post.hook.backup.velero.io/container: fsfreeze
+        post.hook.backup.velero.io/command: '["/sbin/fsfreeze", "--unfreeze", "/var/log/nginx"]'
+    spec:
+      volumes:
+        - name: nginx-logs
+          persistentVolumeClaim:
+            claimName: nginx-logs
+      containers:
+        - image: nginx:1.17.6
+          name: nginx
+          ports:
+            - containerPort: 80
+          volumeMounts:
+            - mountPath: /var/log/nginx
+              name: nginx-logs
+              readOnly: false
+        - image: ubuntu:bionic
+          name: fsfreeze
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: /var/log/nginx
+              name: nginx-logs
+              readOnly: false
+          command:
+            - /bin/bash
+            - -c
+            - sleep infinity
+EOk
+
+  kubectl apply -f base.yaml
+
+  wait_for_pod "app=nginx" "nginx-example" 300
+
+  # Create a test backup
+  velero backup create nginx-backup --include-namespaces $TEST_NAMESPACE
+
+  # Wait for backup to complete
+  echo "Waiting for backup to complete..."
+  sleep 200
```

Comment:
Rather than waiting a static 200 seconds every time, any way we can add a retry and check if its completed? i.e check if completed, wait 30 seconds, and do this in a loop say 10 times? the jenkins tests and keycloak-fips tests have some examples potentially which could be copied here, but i'm not sure of the cmd to check the backup status, assume its just a veleroctl command?

Reply:
there is a `-w` flag for waiting the restore process to be completed, I'm going to try it with this.